src/routes/admin/domains/editDomain.ts

```diff
@@ -3,61 +3,93 @@
+        cert = amtDomain.ProvisioningCert
+        domainPwd = amtDomain.ProvisioningCertPassword
+        amtDomain.ProvisioningCert = `${amtDomain.Name}_CERT_KEY`
+        amtDomain.ProvisioningCertPassword = `${amtDomain.Name}_CERT_PASSWORD_KEY`
       }
-    }
-
-    let errorReason
-    // SQL Query > Insert Data
-    const results = await domainsDb.updateDomain(amtDomain).catch((reason) => {
-      errorReason = reason
-    })
-
-    if (!errorReason && amtDomain.ProvisioningCertPassword && results > 0) {
-      // store the password sent into Vault
-      if (req.secretsManager) {
-        log.debug('Store in vault')
-        await req.secretsManager.writeSecretWithKey(`${EnvReader.GlobalEnvConfig.VaultConfig.SecretsPath}certs/${amtDomain.Name}/PROVISIONING_CERT`, 'PROVISIONING_CERT_PASSWORD_KEY', pwdBefore)
-        log.debug('Password written to vault')
-        amtDomain.ProvisioningCertPassword = 'PROVISIONING_CERT_PASSWORD_KEY'
-      } else {
-        log.debug('No secrets manager configured. Storing in DB.')
-        log.debug('Password will be visible in plain text.')
+      // SQL Query > Insert Data
+      const results = await domainsDb.updateDomain(amtDomain)
+      if (results) {
+        // Delete the previous values of cert and password in vault and store the updated values
+        if (req.secretsManager && (newDomain.provisioningCert != null || newDomain.provisioningCertPassword != null)) {
+          const data = { data: {} }
+          await req.secretsManager.deleteSecretWithPath(`${EnvReader.GlobalEnvConfig.VaultConfig.SecretsPath}certs/${amtDomain.Name}`)
+          data.data[`${amtDomain.Name}_CERT_KEY`] = cert
+          data.data[`${amtDomain.Name}_CERT_PASSWORD_KEY`] = domainPwd
+          await req.secretsManager.writeSecretWithObject(`${EnvReader.GlobalEnvConfig.VaultConfig.SecretsPath}certs/${amtDomain.Name}`, data)
+          log.debug(`Updated AMT Domain : ${amtDomain.Name} in vault`)
+        }
+        res.status(204).end()
       }
     }
-
-    if (typeof results === 'undefined' || results === null || errorReason) {
-      res.status(500).end(`${errorReason}`)
+  } catch (error) {
+    log.error(`Failed to update AMT Domain : ${amtDomain.Name}`, error)
+    if (error instanceof RPSError) {
+      res.status(400).json(API_RESPONSE(400, null, error.name, error.message)).end()
     } else {
-      if (results > 0) {
-        res.status(200).end(DOMAIN_SUCCESSFULLY_UPDATED(amtDomain.Name))
-      } else {
-        res.status(404).end(DOMAIN_NOT_FOUND(amtDomain.Name))
-      }
+      res.status(500).json(API_RESPONSE(500, null, null, API_UNEXPECTED_EXCEPTION(`UPDATE ${amtDomain.Name}`))).end()
     }
-  } catch (error) {
-    log.error(error)
-    res.end(DOMAIN_ERROR(amtDomain.Name))
   }
 }
+
+function getUpdatedData (newDomain: any, oldDomain: AMTDomain): AMTDomain {
+  const amtDomain: AMTDomain = {} as AMTDomain
+  amtDomain.Name = newDomain.profileName
+
+  if (newDomain.domainSuffix == null) {
```

Comment:
you probably can use the Coalesce Nullish operator (??) here:
`amtDomain.DomainSuffix  = oldDomain.DomainSuffix ?? newDomain.domainSuffix`
will simplify the code a bit

Reply:
done